build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
+@@ -120,6 +120,8 @@ public class MainSettings extends PreferenceFragmentCompat
+             findPreference(PREF_PRIVACY).setTitle(R.string.prefs_privacy_security);
+         }
+ 
++        createBromitePreferences();
++
+         cachePreferences();
+ 
+         updatePasswordsPreference();
+@@ -274,4 +276,9 @@ public class MainSettings extends PreferenceFragmentCompat
+             }
+         };
+     }
++
++    private void createBromitePreferences() {
++        SettingsUtils.inflatePrefsAddon(this, "main_preferences_addon_",
++            R.xml.class.getFields());
++    }
+ }
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java
+--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/SettingsActivity.java
+@@ -274,9 +274,13 @@ public class SettingsActivity extends ChromeBaseAppCompatActivity
+             finish();
+             return true;
+         } else if (item.getItemId() == R.id.menu_id_general_help) {
+-            HelpAndFeedbackLauncherImpl.getInstance().show(this,
+-                    getString(R.string.help_context_settings), Profile.getLastUsedRegularProfile(),
+-                    null);
++            if (mainFragment instanceof SettingsUtils.ISupportHelpAndFeedback) {
++                ((SettingsUtils.ISupportHelpAndFeedback)mainFragment).onHelpAndFeebackPressed();
++            } else {
++                HelpAndFeedbackLauncherImpl.getInstance().show(this,
++                        getString(R.string.help_context_settings), Profile.getLastUsedRegularProfile(),
++                        null);
++            }
+             return true;
+         }
+         return super.onOptionsItemSelected(item);
+diff --git a/chrome/android/java_sources.gni b/chrome/android/java_sources.gni
+--- a/chrome/android/java_sources.gni
++++ b/chrome/android/java_sources.gni
+@@ -19,6 +19,7 @@ import("//components/feed/features.gni")
+ import("//components/offline_pages/buildflags/features.gni")
+ import("//components/omnibox/browser/test_java_sources.gni")
+ import("//device/vr/buildflags/buildflags.gni")
++import("//components/user_scripts/android/java_sources.gni")
+ 
+ chrome_java_sources += public_autofill_assistant_java_sources
+ 
+@@ -51,3 +52,5 @@ if (enable_arcore) {
+     "java/src/org/chromium/chrome/browser/vr/ArDelegateImpl.java",
+   ]
+ }
++
++chrome_java_sources += userscripts_java_sources
+diff --git a/chrome/android/proguard/main.flags b/chrome/android/proguard/main.flags
+--- a/chrome/android/proguard/main.flags
++++ b/chrome/android/proguard/main.flags
+@@ -63,3 +63,10 @@
+ -keep,allowobfuscation public class ** extends org.chromium.chrome.browser.base.SplitCompat*$Impl {
+   public <init>();
+ }
++
++# Keep fields name for all R.xml
```

Comment:
Shall we move this to its own separate patch? It seems to have benefits that are also outside of user scripts support.

Reply:
fixed in https://github.com/bromite/bromite/pull/857/commits/8c164138391082350f20493a0ca5c4926887372c